src/tracked_struct/struct_map.rs

```diff
@@ -100,7 +100,15 @@ where
     }
 
     pub fn validate<'db>(&'db self, runtime: &'db Runtime, id: Id) {
-        let mut data = self.map.get_mut(&id).unwrap();
+        Self::validate_in_map(&self.map, runtime, id)
+    }
+
+    pub fn validate_in_map<'db>(
+        map: &'db FxDashMap<Id, Alloc<Value<C>>>,
+        runtime: &'db Runtime,
+        id: Id,
+    ) {
```

Comment:
I think `validate_in_map` could be private. I don't see it being called from a macro. 

It would be great if you could add a coment to this method. You must heave learned a lot when fixing this bug, let's preserve and share this knowledge